xbmc/windows/GUIMediaWindow.cpp

```diff
@@ -727,30 +713,36 @@ bool CGUIMediaWindow::GetDirectory(const std::string &strDirectory, CFileItemLis
 // \brief Set window to a specific directory
 // \param strDirectory The directory to be displayed in list/thumb control
 // This function calls OnPrepareFileItems()
-bool CGUIMediaWindow::Update(const std::string &strDirectory, bool updateFilterPath /* = true */)
+bool CGUIMediaWindow::Update(const std::string &_arg, bool updateFilterPath /* = true */)
```

Comment:
Then I'll rename it strDirectoryOrEmpty or something unless you have a better suggestion. If this parameter was "The current directory path" I'd agree with you, but it isn't. This argument has at least 3 different meanings here: "?" some magic value for doing nothing, empty string which means get the root, or the actual path.